### DISCUSSION TITLE
Questions about way hutch connect with Rabbit

### DISCUSSION THREAD

**Discussion by ricsdeol** [gocardless/hutch#367] [Q&A]
First of all, congratulations for the excellent project.

I had a discussion with my team (@gtkatakura) about this, because we had a worker that publish a message after finish the process, and it is called many times.  We have an attention that this opens many connections with Rabbit. And we have some questions.

When will this connection be closed?

```ruby 
Hutch.connect
Hutch.publish('routing.key', subject: 'payment', action: 'received') 
```

Should we close the connection `Hutch.disconnect` after publishing ? If yes, is there a way best practice  do like this:
```ruby
Hutch.connect
Hutch.publish('routing.key', subject: 'payment', action: 'received') 
Hutch.disconnect
``` 

Is the connection of `Hutch.connect`  the same of consumers?

Each consumer has your connection or be shared by process?

**Comment by akabiru** [comment]
Insightful Q @ricsdeol , (My thoughts)

Given hutch will reuse connections [AFAICS](https://github.com/ruby-amqp/hutch/blob/master/lib/hutch.rb#L46), I reckon there's no need to disconnect manually? Given it's expensive to set up the connection initially. 

I suspect the case where disconnecting would be useful is when there's more than one broker, that is not part of a cluster.
